website/blog/2024-04-04-contributor-guidelines/index.mdx

```diff
@@ -0,0 +1,86 @@
+
+To get started, contributors need to find an issue to work on. Make sure you have open issues at the ready that are tagged to cater to all potential contributors. If a junior developer wants to learn a new language through your project make sure `good first issues` are easy to find. If more ambitious, senior types, are ready to make an impact, make sure it’s clear which issues you would like them to consider, pointing them to a [current milestone](https://github.com/glasskube/glasskube/milestones) or GitHub [project](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects). 
+
+## Define contribution paths
+
+Make clear what kind of paths there are to contribute to. Which areas of the project need help? Leverage GitHub tags to segment each project domain. In our case out project has different components such as a `GUI`, a `CLI` and `Package Controller`, `Docs`, a `UI` and the repo itself. 
+
+## Community engagement
+
+Ok, so the contributor knows that there are open issues ranging in different levels of complexity and they understand the different areas they can contribute to. How can they let you know they are ready to contribute to the project? What if they have additional questions?
+Make it clear how you prefer them to communicate with you, through GitHub, [Discord](https://discord.gg/SxH6KUCGH7), or Slack. 
+
+> Try to not only limit communication to the GitHub issue itself. For many Open Source contributors the fun is meeting other like minded people to collaborate and share. Not to say they everybody want that, but make sure you invite contributors to a community, since they are not part of it. 
+
+![Community engagement](https://github.com/glasskube/glasskube/assets/38757612/a582fac4-1bd9-4db2-b3c3-61c677befa72)
+
+## Contribution workflow
+
+Probably one of the most important parts of the Contributor guidelines is for contributors to have clear instructions and answers to the following questions
+
+Detailing the step-by-step process for contributing code changes: 🪜
+
+- Issue selection and assignment.
+- Forking the repository.
+- Branching, coding, and committing conventions.
+- Pull request submission and review.
+- Handling feedback and iteration.
+- Merging contributions into the main codebase.
+- Addressing best practices for commit messages and keeping forks up-to-date.
+
+
+## Development guidelines
+
+Each open-source project aims to solve a different problem and build on a certain stack that might be new to some contributors. Assume they know nothing about your tool and precisely explain the project’s building blocks.
+
+Focus on informing how the code base is built how each component fits together and describing setup environments.  Ideally, contributors should be able to build local versions of the components they contribute to and validate their work. 
+
+If you only find out during the PR review process that it wasn’t clear to the developer, it’s already too late. 😡
```

Comment:
Yes, but also not too late to adapt for future contributors. I think that could be mentioned – if someone asks, it's a sign that maybe it's not good enough yet – something like that.

Reply:
Sorry, I don't fully understand what you mean here.

Reply:
I just thought "it's already too late" is a bit harsh. It's not too late, it's just an indicator that something was not clear enough, and should be improved.

Reply:
ahhhhh got it, true, let me change it